File: NetFramework/FlexGrid/CS/AdvancedFiltering/InheritedFilterEditor.cs
using C1.Win.C1FlexGrid;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace AdvancedFiltering
{
    public partial class InheritedFilterEditor : ColumnFilterEditor
    {
        private InheritedFilter _inheritedFilter;

        public InheritedFilterEditor()
        {
            InitializeComponent();
        }

        public override void Initialize(C1FlexGridBase grid, int columnIndex, IC1ColumnFilter filter)
        {
            base.Initialize(grid, columnIndex, filter);
            _weekdayFilterEditor.Initialize(grid, columnIndex, filter);
            _inheritedFilter = (InheritedFilter)filter;

            if (_inheritedFilter.WeekdayFilter.IsActive || !_inheritedFilter.IsActive)
            {
                ActiveEditor = _weekdayFilterEditor;
            }
        }

        private void _btnWeekdayFilter_Click(object sender, EventArgs e)
        {
            ActiveEditor = _weekdayFilterEditor;
        }

        public override void ApplyChanges()
        {
            base.ApplyChanges();

            if (ActiveEditor != _weekdayFilterEditor)
            {
                _inheritedFilter.WeekdayFilter.Reset();
            }
        }

        public override IC1ColumnFilterEditor ActiveEditor
        {
            get
            {
                return _weekdayFilterEditor.Visible ? _weekdayFilterEditor : base.ActiveEditor;
            }
            set
            {
                base.ActiveEditor = value;
                _weekdayFilterEditor.Visible = value == _weekdayFilterEditor;
            }
        }
    }
}

File: NetFramework/FlexGrid/VB/AdvancedFiltering/InheritedFilterEditor.vb
Imports C1.Win.FlexGrid
Imports System
Imports System.Windows.Forms

Namespace AdvancedFiltering
    Partial Public Class InheritedFilterEditor
        Inherits ColumnFilterEditor
        Private _inheritedFilter As InheritedFilter

        Public Sub New()
            InitializeComponent()
        End Sub

        Public Overrides Sub Initialize(ByVal grid As C1FlexGridBase, ByVal columnIndex As Integer, ByVal filter As IC1ColumnFilter)
            MyBase.Initialize(grid, columnIndex, filter)
            _weekdayFilterEditor.Initialize(grid, columnIndex, filter)
            _inheritedFilter = CType(filter, InheritedFilter)

            If _inheritedFilter.WeekdayFilter.IsActive OrElse Not _inheritedFilter.IsActive Then
                ActiveEditor = _weekdayFilterEditor
            End If
        End Sub

        Private Sub _btnWeekdayFilter_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnWeekdayFilter.Click
            ActiveEditor = _weekdayFilterEditor
        End Sub

        Public Overrides Sub ApplyChanges()
            MyBase.ApplyChanges()

            If ActiveEditor IsNot _weekdayFilterEditor Then
                _inheritedFilter.WeekdayFilter.Reset()
            End If
        End Sub

        Public Overrides Property ActiveEditor As IC1ColumnFilterEditor
            Get
                Return If(_weekdayFilterEditor.Visible, _weekdayFilterEditor, MyBase.ActiveEditor)
            End Get
            Set(ByVal value As IC1ColumnFilterEditor)
                MyBase.ActiveEditor = value
                _weekdayFilterEditor.Visible = value Is _weekdayFilterEditor
            End Set
        End Property
    End Class
End Namespace
